python/cuda_parallel/tests/test_reduce.py

```diff
@@ -13,7 +13,6 @@
 import numba.cuda
 import numba.types
 import cuda.parallel.experimental as cudax
-from cuda.parallel.experimental import _iterators
```

Comment:
Note: I removed all usages of the private module from the tests as discussed in https://github.com/NVIDIA/cccl/pull/2788#discussion_r1872403518. To follow up, do we want an additional test for testing pointer handling in `TransformIterator`?